### ISSUE TITLE
[For Gadgetbridge User]Invalid API Keys

### ISSUE THREAD

**Issue by SteveZMTstudios** [gelin/weather-notification#71]
Recently This Application seems no longer usable and the upstream weather provider(openweathermap.org) responde 401 invalid key.

Is this APPID expired?

![Screenshot_2024-08-16-03-01-35-244_ru.gelin.android.weather.notification.jpg](https://github.com/user-attachments/assets/5ae6f2f8-a813-4135-b549-9a415243d3ae)

**Comment by doodhout** [comment]
OpenWeatherMap has discontinued their API version 2 which this app uses, you need to register at OWM and provide credit card info to have access to their API version 3 but the first 1k calls per day are free and the price of subsequent calls that day are cheap, plus you can limit the daily rate of calls to not go over the free limit: https://openweathermap.org/full-price#onecall

That being said, does this app support API 3 of OWM?

EDIT: The URL that this app uses to consult OWM has 2.5 hardcoded into it (for API 2.5, the latest version of API 2), which means that this app will likely not be able to use API 3.0 even if you subscribed to One Call API 3.0.

TL;DR: I'm guessing this app needs to be updated to support API 3.0

EDIT 2: If you are here because you rely on this app to feed weather data to GadgetBridge, there are other apps supported by GadgetBridge: https://gadgetbridge.org/internals/development/weather-support

**Comment by SteveZMTstudios** [comment]
> OpenWeatherMap has discontinued their API version 2 which this app uses, you need to register at OWM and provide credit card info to have access to their API version 3 but the first 1k calls per day are free and the price of subsequent calls that day are cheap, plus you can limit the daily rate of calls to not go over the free limit: https://openweathermap.org/full-price#onecall
> 
> That being said, does this app support API 3 of OWM?
> 
> EDIT: The URL that this app uses to consult OWM has 2.5 hardcoded into it (for API 2.5, the latest version of API 2), which means that this app will likely not be able to use API 3.0 even if you subscribed to One Call API 3.0.
> 
> TL;DR: I'm guessing this app needs to be updated to support API 3.0

Thank you for these ways to solve this problem.  The real problem is in my regioni can't register an credit card (as a student I have no source of income ). But this is no longer relevant to this issue.

**Comment by paolo-caroni** [comment]
Please leave it open, for other people like me that use gadgetbridge :-)

**Comment by baianohak** [comment]
How to make it works? Can you update the app. Please. I don't know what gadget bridge is.

**Comment by paolo-caroni** [comment]
@baianohak I have moved to Breezy Wheather (that is compatible with gadgetbridge) but if you don't use a smartwatch/smartband/external device... You can use any weather app. This app seems to be unmaintained

**Comment by SteveZMTstudios** [comment]
Oh, Gadgetbridge is an open source app to bind watchs with phone.Some watches supports to show weather data in their surfaces. Gadgetbridges use other weather app to get weather forecast data. This app is one of them.But it's app seems not updated and its api key is expired.  So this issue is to use to give that GadgetBridge users another way to update weather data.

**Comment by Guintolli** [comment]
Hello,
I have been using this app for many years, paired with GadgetBridge. 
This weekend, I encountered the 401 error (strange that I wasn't affected earlier). 
While waiting for an update from the developer, I created a fork to only change the OpenWeatherMap URL and rebuilt an Android APK (I should mention I don’t have any particular expertise). If this can help others, my repo is public ;)

**Comment by freecycler** [comment]
> I created a fork to only change the OpenWeatherMap URL and rebuilt an Android APK

1. A link would have been helpful
2. If anyone else wants it, I found the APK at https://github.com/Guintolli/weather-notification/releases
3. It still doesnt work. I would have left this comment there, but issues are disabled on that repo.

**Comment by Guintolli** [comment]
> > I created a fork to only change the OpenWeatherMap URL and rebuilt an Android APK
> 
>     1. A link would have been helpful
> 
>     2. If anyone else wants it, I found the APK at https://github.com/Guintolli/weather-notification/releases
> 
>     3. It still doesnt work. I would have left this comment there, but issues are disabled on that repo.

@freecycler : Hi,
I just activated the issues functionality on the repo.
Don't hesitate to describe your issue, I'll see if I can help you.
I use the new apk (new url) without issue since one week

**Comment by freecycler** [comment]
> 3. It still doesnt work. I would have left this comment there, but issues are disabled on that repo.

After a little more digging, maybe I should rephrase that... it still doesn't work, **_unless you give them your credit card details_**.

So thanks for the offer to help, but if it needs a credit card on file, I'll not be giving them that. Have you found any other solution?

**Comment by Guintolli** [comment]
> > 3. It still doesnt work. I would have left this comment there, but issues are disabled on that repo.
> 
> After a little more digging, maybe I should rephrase that... it still doesn't work, **_unless you give them your credit card details_**.
> 
> So thanks for the offer to help, but if it needs a credit card on file, I'll not be giving them that. Have you found any other solution?

Unfortunately, OpenWeatherMap has changed the management of some of their free subscriptions. Indeed, it is mandatory to register a credit card to use the APIs we need (even though the first 1000 calls per day are free). I used a virtual credit card (with a limit of 1€) to avoid providing my real card details
